controller/src/enhancements/player.rs

```diff
@@ -261,6 +263,21 @@ impl Enhancement for PlayerESP {
             }
         };
 
+        let local_controller = ctx.cs2_entities.get_local_player_controller()?;
+        if local_controller.is_null()? {
+            return Ok(());
+        }
+
+        let local_pawn = ctx
+            .cs2_entities
+            .get_by_handle(&local_controller.reference_schema()?.m_hPlayerPawn()?)?
+            .context("missing local player pawn")?
+            .entity()?
+            .read_schema()?;
+
+        let localpos = nalgebra::Vector3::<f32>::from_column_slice(&local_pawn.m_vOldOrigin()?);
```

Comment:
Local position should be the position of the currently observing pawn (`observice_entity_handle`). When controlling an entity thus would be the local player pawn, but when observing it must be equal.  
I may would suggest updating the local position where the comment `/* current pawn we control/observe */` currently is located.  
Be aware, that we might never find the observing entity. Therefore I would change `local_pos` to `Option<nalgebra::Vector3<f32>>`

Reply:
`        for entity_identity in ctx.cs2_entities.all_identities() {
            if entity_identity.handle::<()>()?.get_entity_index() == observice_entity_handle {
                /* current pawn we control/observe */
                let localpos = nalgebra::Vector3::<f32>::from_column_slice(&local_pawn.m_vOldOrigin()?);
                self.local_pos = Some(localpos);
                continue;
            }`
like this? i didnt understand, but the distance keeps working when im observing someone: 
![image](https://github.com/Valthrun/Valthrun/assets/80542190/1bf3c03e-f6f7-4b0a-a047-4ec9212c041c)

Reply:
Yep exactly like that :)